src/scenes/ObservationsRouter/plot/AdHocObservationDetails.tsx

```diff
@@ -101,12 +134,123 @@ export default function AdHocObservationDetails(): JSX.Element {
     }
   }, [navigate, monitoringPlot]);
 
+  useEffect(() => {
+    if (!allSpecies && selectedOrganization.id !== -1) {
+      dispatch(requestSpecies(selectedOrganization.id));
+    }
+  }, [dispatch, allSpecies, selectedOrganization]);
+
+  useEffect(() => {
+    const speciesWithNoIdMap = _.uniqBy(
+      (monitoringPlot?.species || []).filter((sp) => !sp.speciesId),
+      'speciesName'
+    ).map((sp) => sp.speciesName || '');
+
+    setUnrecognizedSpecies(speciesWithNoIdMap);
+    if (speciesWithNoIdMap.length > 0) {
+      setShowPageMessage(true);
+    } else {
+      setShowPageMessage(false);
+    }
+  }, [monitoringPlot]);
+
+  useEffect(() => {
+    if (matchResponse?.status === 'success' && matchResponse?.data && matchResponse.data.length > 0) {
+      // Force reload page to show updated data
+      window.location.reload();
```

Comment:
Instead of doing this reload you can pass the reload function as an attribute and call it here.
In src/scenes/ObservationsRouter/index.tsx:

`      <Route
        path={'/:plantingSiteId/results/:observationId/adHocPlot/:monitoringPlotId'}
        element={<AdHocObservationDetails reload={reload}/>}
      />`